src/ProjectOrigin.RequestProcessor/Models/RequestId.cs

```diff
@@ -0,0 +1,21 @@
+namespace ProjectOrigin.RequestProcessor.Models;
+
+public record RequestId(byte[] RequestHash)
+{
+    public virtual bool Equals(RequestId? other)
+    {
+        if (other is null) return false;
+
+        return RequestHash.SequenceEqual(other!.RequestHash);
+    }
+
+    public override int GetHashCode()
+    {
+        var hc = RequestHash.Length;
+        foreach (int val in RequestHash)
+        {
+            hc = unchecked(hc * 314159 + val);
```

Comment:
I assume 314159 is a random number typed by the programmer - but I might be wrong :-)

Reply:
It is a magic hash number used many places in dotnet for this purpose.